### DISCUSSION TITLE
I don't understand "Preparing Dataset (With Caching)"

### DISCUSSION THREAD

**Discussion by FugueSegue** [d8ahazard/sd_dreambooth_extension#776] [Q&A]
With improvements to the extension, I've noticed changes. The one that has me puzzled is something that happens at the beginning of training. What is "Preparing Dataset (With Caching)" and how does it come up with the number of items that are cached?

For example, I'm training a model that has 36 dataset images. I set the Epochs to 100 and the Class Images Per Instance Image to 10. When I start training, it says that the Instance Bucket 0 is 36 and that the Class Bucket 0 is 9800. Then I sit and wait a long time for 9836 things to cache. It's clear that the Instance Bucket number corresponds to the number of dataset images. But I don't understand why the Class Bucket is 9800. Where does that number come from?

And a related questions: is 10 class images per instance image a large amount? Should I just use 1 class image per instance image in most training situations?

**Comment by FugueSegue** [comment]
Well, I discovered the origin of the number 9800. It's the number of classification images I have already generated in that folder.

But I'm still puzzled. Why does it cache all of the classification images present in the folder when I only need (Dataset Images * Class Images Per Instance Images) for a training? Or is there something I don't understand?

**Reply by Thomas-MMJ** [reply]
> Why does it cache all of the classification images present in the folder when I only need (Dataset Images * Class Images Per Instance Images) for a training? Or is there something I don't understand?

Probably the code just doesn't consider the case of their being more images available than needed, and so simply does all files in the folder.
